packages/cli/src/modules/insights/insights.controller.ts

```diff
@@ -1,16 +1,43 @@
-import type { InsightsSummary } from '@n8n/api-types';
+import { ListInsightsWorkflowQueryDto } from '@n8n/api-types';
+import type { InsightsSummary, InsightsByTime, InsightsByWorkflow } from '@n8n/api-types';
 
-import { Get, GlobalScope, RestController } from '@/decorators';
+import { Get, GlobalScope, Query, RestController } from '@/decorators';
+import { paginationListQueryMiddleware } from '@/middlewares/list-query/pagination';
+import { sortByQueryMiddleware } from '@/middlewares/list-query/sort-by';
+import { AuthenticatedRequest } from '@/requests';
 
 import { InsightsService } from './insights.service';
 
 @RestController('/insights')
 export class InsightsController {
+	private readonly nbDaysFilteredInsights = 14;
+
 	constructor(private readonly insightsService: InsightsService) {}
 
 	@Get('/summary')
 	@GlobalScope('insights:list')
 	async getInsightsSummary(): Promise<InsightsSummary> {
 		return await this.insightsService.getInsightsSummary();
 	}
+
+	@Get('/by-workflow', { middlewares: [paginationListQueryMiddleware, sortByQueryMiddleware] })
+	@GlobalScope('insights:list')
```

Comment:
Let's add a controller test making sure the permissions work. I forgot that for the `summary` route as well.